### PR TITLE
feat(CAD): Add flow metrics for connect another device.

### LINKED ISSUE #4783: Emit flow events for /connect_another_device
@philbooth showed me how easy it was to instrument the front end to emit flow events. We never did so with Connect Another Device phase 1 because we emit events to DataDog using the experiments infrastructure. As soon as we [remove the experiment](https://github.com/mozilla/fxa-content-server/pull/4781), we are going to lose DataDog metrics related to how well this feature is performing, and we won't be reporting metrics to re:dash. :(

### PR DESCRIPTION
**Pull request by shane-tomlinson**
I'm not totally happy with this solution, there's a lot of duplication
between this and the experiment metrics, it seems like we should
be able to unify. That's for another day.

fixes #4783

@philbooth - r?

### CODE REVIEW

File: app/tests/spec/views/connect_another_device.js [hunk 1]
```diff
@@ -116,14 +139,20 @@ define(function (require, exports, module) {
...
         });
 
         it('shows a sign in button with the appropriate link, logs appropriately', () => {
           assert.lengthOf(view.$('#signin'), 1);
           testIsNotified('connectAnotherDevice.signedin.false');
           testIsNotified('connectAnotherDevice.signin.eligible');
           testIsNotified('connectAnotherDevice.signin_from.fx_desktop');
+          testIsFlowEventLogged('signedin.false');
+          testIsFlowEventLogged('signin.eligible');
+          testIsFlowEventLogged('signin_from.fx_desktop');
```
Comment (vladikoff) [inline app/tests/spec/views/connect_another_device.js]: Do we need to update the flow even docs to include these new events?

File: app/scripts/templates/connect_another_device.mustache [hunk 2]
```diff
@@ -64,7 +64,7 @@
       {{/isFirefoxIos}}
     {{/canSignIn}}
     <div class="links">
-      <a href="/connect_another_device/why">{{#t}}Why is this required?{{/t}}</a>
+      <a data-flow-event="link.why" href="/connect_another_device/why">{{#t}}Why is this required?{{/t}}</a>
```
Comment (vladikoff) [inline app/scripts/templates/connect_another_device.mustache]: I wonder if are overloading the flow event pipeline. I think before we were talking about how flowEvents are only a selected number of events. If we keep adding more and more then queries will be even slower. thoughts on this @philbooth ?

Comment (philbooth) [inline app/scripts/templates/connect_another_device.mustache]: But it's THE CLOUD! 😄

I see your point but I think we're good. At some point soon, we'll implement automated expiry for the flow data like we have for the activity events and that should address most of the performance concerns. As long as they're useful events, the more the merrier.

Comment (vladikoff) [inline app/scripts/templates/connect_another_device.mustache]: Ok! sounds good

File: app/tests/spec/views/connect_another_device.js (hunk 1 above)
Comment (philbooth) [inline app/tests/spec/views/connect_another_device.js]: Those event docs are next in the line of fire on my auto-generate-all-the-docs mission.

**Comment by philbooth** [comment]
@shane-tomlinson, this looks good to me! r+

File: app/tests/spec/views/connect_another_device.js (hunk 1 above)
Comment (shane-tomlinson) [inline app/tests/spec/views/connect_another_device.js]: @philbooth - should I add them to the doc manually, or wait for auto-generation?

**Pull request merged**